inc/core/limited_offers.php

```diff
@@ -0,0 +1,336 @@
+	public function is_deal_active( $slug ) {
+
+		if ( empty( $slug ) || ! array_key_exists( $slug, $this->timelines ) ) {
+			return false;
+		}
+
+		return $this->check_date_range( $this->timelines[ $slug ]['start'], $this->timelines[ $slug ]['end'] );
+	}
+
+	/**
+	 * Get the remaining time for the deal in a human readable format.
+	 *
+	 * @param string $slug Slug of the deal.
+	 * @return string Remaining time for the deal.
+	 */
+	public function get_remaining_time_for_deal( $slug ) {
+		if ( empty( $slug ) || ! array_key_exists( $slug, $this->timelines ) ) {
+			return '';
+		}
+
+		try {
+			$end_date     = new DateTime( $this->timelines[ $slug ]['end'], new DateTimeZone( 'GMT' ) );
+			$current_date = new DateTime( 'now', new DateTimeZone( 'GMT' ) );
+			$diff         = $end_date->diff( $current_date );
+
+			if ( $diff->days > 0 ) {
+				return $diff->format( '%a days' );
+			}
+
+			if ( $diff->h > 0 ) {
+				return $diff->format( '%h hours' );
+			}
+
+			if ( $diff->i > 0 ) {
+				return $diff->format( '%i minutes' );
+			}
+
+			return $diff->format( '%s seconds' );
+		} catch ( Exception $e ) {
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+				error_log( $e->getMessage() ); // phpcs:ignore
+			}
+		}
+
+		return '';
+	}
+
+	/**
+	 * Check if the current date is in the range of the offer.
+	 *
+	 * @param string $start Start date.
+	 * @param string $end   End date.
+	 *
+	 * @throws Exception When date is invalid.
+	 */
+	public function check_date_range( $start, $end ) {
+
+		$start_date   = new DateTime( $start, new DateTimeZone( 'GMT' ) );
+		$end_date     = new DateTime( $end, new DateTimeZone( 'GMT' ) );
+		$current_date = new DateTime( 'now', new DateTimeZone( 'GMT' ) );
+
+		return $start_date <= $current_date && $current_date <= $end_date;
+	}
+
+	/**
+	 * Get the localized data for the plugin.
+	 *
+	 * @return array Localized data.
+	 */
+	public function get_localized_data() {
+		return array_merge(
+			array(
+				'active'        => $this->is_active(),
+				'dealSlug'      => $this->get_active_deal(),
+				'remainingTime' => $this->get_remaining_time_for_deal( $this->get_active_deal() ),
+				'urgencyText'   => 'Hurry Up! Only ' . $this->get_remaining_time_for_deal( $this->get_active_deal() ) . ' left',
+			),
+			$this->offer_metadata
+		);
+	}
+
+	/**
+	 * Disable the notification via ajax.
+	 *
+	 * @return void
+	 */
+	public function disable_notification_ajax() {
+		if ( ! isset( $_POST['nonce'] ) || ! wp_verify_nonce( sanitize_key( $_POST['nonce'] ), 'dismiss_themeisle_event_notice_neve' ) ) {
+			wp_die( 'Invalid nonce! Refresh the page and try again.' );
+		}
+
+		// We record the time and the plugin of the dismissed notification.
+		update_option( $this->wp_option_dismiss_notification_key_base . $this->active, 'neve_' . $this->active . '_' . current_time( 'Y_m_d' ) );
+		wp_die( 'success' );
+	}
+
+	/**
+	 * Render the dashboard banner.
+	 *
+	 * @return void
+	 */
+	public function render_dashboard_banner() {
```

Comment:
The dashboard banner is always visible, without checking if it's the BF date. Is that the desired behavior? I just activated Neve and the notification is showing. Also, there's no CTA 
<img width="1144" alt="Screenshot 2023-10-18 at 14 44 14" src="https://github.com/Codeinwp/neve/assets/9929553/0ef4218c-f764-45f8-8792-2489ee571cb2">

Reply:
Thanks for the CTA. Regarding the time, the start date is the test. It will be changed after QA approval.